test/unix/concurrent.ml

```diff
@@ -0,0 +1,29 @@
+open Common
+
+let index_name = Filename.concat "_tests" "unix.concurrent"
+
+let create_index () =
+  Logs.debug (fun l -> l "pid = %d" (Unix.getpid ()));
+  let _t = Index.v ~fresh:true ~log_size index_name in
+  ignore (tbl index_name)
+
+let concurrent_runs () =
+  if Sys.os_type = "Win32" then ()
+  else
+    let exn =
+      Failure
+        ("Lock didn't succeed: file " ^ index_name ^ "/index/.lock is present")
+    in
+    let pid = Unix.fork () in
+    if pid <> 0 then (
+      Unix.sleepf 0.001;
```

Comment:
this might be a bit fragile, but I guess that's good enough for now. We can restructure this later if we see that this the tests is not deterministic enough.

Reply:
Now that we have a `Private` functor abstraction, could we add a hook for the creation of the lock file to test this more consistently?

It would be nice to have these tests merged so that they're not a maintenance burden to you :slightly_smiling_face:

Reply:
Thanks for the idea, I removed all calls to `Unix.sleep`.